config/_templates/dataset/carla_garage/town10.yaml

```diff
@@ -0,0 +1,353 @@
+        output_name: features
+        mapspec: "waypoints_path[i] -> features[i]"
+        func:
+          _target_: rbyte.io.GeoDataFrameBuilder
+          srid: 0
+          columns: [heading, timestamp, geometry]
+
+      - _target_: pipefunc.PipeFunc
+        renames:
+          input: features
+        output_name: features_converted
+        mapspec: "features[i] -> features_converted[i]"
+        func:
+          _target_: rbyte.io.DataFrameQuery
+          query: |
+            SELECT * FROM self ORDER BY timestamp
+
+      - _target_: pipefunc.PipeFunc
+        renames:
+          input: features_converted
+        output_name: waypoints
+        mapspec: "features_converted[i] -> waypoints[i]"
+        func:
+          _target_: rbyte.io.WaypointBuilder
+          length: 10
+          columns:
+            points: geometry
+            output: waypoints
+
+      - _target_: pipefunc.PipeFunc
+        output_name: data
+        mapspec: "meta[i], waypoints[i] -> data[i]"
+        func:
+          _target_: pipefunc.helpers.collect_kwargs
+          parameters: [meta, waypoints]
+          function_name: aggregate
+
+      - _target_: pipefunc.PipeFunc
+        renames:
+          input: data
+        output_name: aligned
+        mapspec: "data[i] -> aligned[i]"
+        func:
+          _target_: rbyte.io.DataFrameAligner
+          separator: .
+          fields:
+            meta:
+              key: _idx_
+              columns:
+                records.speed:
+                  method: interp
+                records.steer:
+                  method: interp
+                records.throttle:
+                  method: interp
+                records.brake:
+                  method: interp
+                records.pos_global:
+                  method: asof
+                  strategy: nearest
+
+            waypoints:
+              key: timestamp
+              columns:
+                heading:
+                  method: asof
+                  strategy: nearest
+                waypoints:
+                  method: asof
+
+      - _target_: pipefunc.PipeFunc
+        renames:
+          input: aligned
+        output_name: with_waypoints_normalized
+        mapspec: "aligned[i] -> with_waypoints_normalized[i]"
+        func:
+          _target_: rbyte.io.WaypointNormalizer
+          srid: 0
+          columns:
+            ego: [meta.pos_global_x, meta.pos_global_y]
+            waypoints: waypoints.waypoints
+            heading: waypoints.heading
+            output: waypoints.waypoints_normalized
+
+      - _target_: pipefunc.PipeFunc
+        renames:
+          input: with_waypoints_normalized
+        output_name: samples
+        mapspec: "with_waypoints_normalized[i] -> samples[i]"
+        func:
+          _target_: rbyte.io.FixedWindowSampleBuilder
+          index_column: meta._idx_
+          every: 1i
+          period: 7i
+          closed: both
+          gather_every: 1
+          length: 8
```

Comment:
if you bump to `rbyte>=0.23.0` this can be simplified somewhat, see https://github.com/yaak-ai/rbyte/blob/main/config/_templates/dataset/carla_garage.yaml#L42-L156